public/app/features/alerting/NewAlertNotificationPage.tsx

```diff
@@ -0,0 +1,132 @@
+  testNotificationChannel: typeof testNotificationChannel;
+}
+
+type Props = OwnProps & ConnectedProps & DispatchProps;
+
+const defaultValues: NotificationChannelDTO = {
+  name: '',
+  type: { value: 'email', label: 'Email' },
+  sendReminder: false,
+  disableResolveMessage: false,
+  frequency: '15m',
+  settings: {
+    uploadImage: config.rendererAvailable,
+    autoResolve: true,
+    httpMethod: 'POST',
+    severity: 'critical',
+  },
+  isDefault: false,
+};
+
+class NewAlertNotificationPage extends PureComponent<Props> {
+  componentDidMount() {
+    this.props.loadNotificationTypes();
+  }
+
+  onSubmit = (data: NotificationChannelDTO) => {
+    /*
+      Some settings can be options in a select, in order to not save a SelectableValue<T>
+      we need to use check if it is a SelectableValue and use its value.
+    */
+    const settings = Object.fromEntries(
+      Object.entries(data.settings).map(([key, value]) => {
+        return [key, value.hasOwnProperty('value') ? value.value : value];
+      })
+    );
+
+    this.props.createNotificationChannel({
+      ...defaultValues,
+      ...data,
+      type: data.type.value,
+      settings: { ...Object.assign(defaultValues.settings, settings) },
+    });
+  };
+
+  onTestChannel = (data: NotificationChannelDTO) => {
+    this.props.testNotificationChannel({
+      name: data.name,
+      type: data.type.value,
+      frequency: data.frequency ?? defaultValues.frequency,
+      settings: { ...Object.assign(defaultValues.settings, data.settings) },
+    });
+  };
+
+  render() {
+    const { navModel, notificationChannels } = this.props;
+
+    /*
+     Need to transform these as we have options on notificationChannels,
+     this will render a dropdown within the select.
+
+    TODO: Memoize?
+   */
+    const selectableChannels: Array<SelectableValue<string>> = notificationChannels.map(channel => ({
+      value: channel.value,
+      label: channel.label,
+      description: channel.description,
+    }));
+
+    return (
+      <Page navModel={navModel}>
+        <Page.Contents>
+          <h2>New Notification Channel</h2>
+          <Form onSubmit={this.onSubmit} validateOn="onChange" defaultValues={defaultValues}>
+            {({ register, errors, control, getValues, watch }) => {
+              const selectedChannel = notificationChannels.find(c => c.value === getValues().type.value);
+
+              return (
+                <NewNotificationChannelForm
+                  selectableChannels={selectableChannels}
+                  selectedChannel={selectedChannel}
+                  onTestChannel={this.onTestChannel}
+                  register={register}
+                  errors={errors}
+                  getValues={getValues}
+                  control={control}
+                  watch={watch}
+                  imageRendererAvailable={config.rendererAvailable}
+                />
+              );
+            }}
+          </Form>
```

Comment:
nit:

Is there any reason for keeping the `<Form />` and the content of the form `<NewNotificationChannelForm />` separated? For me that abstraction feels a bit odd since I need to jump back and forth between different contexts to understand the whole form functionality.

An option would be to keep them together and break out small parts of the form into its own components or smaller render functions.

Reply:
We have the same approach here: https://github.com/grafana/grafana/blob/564a34a9c8fdfbb66c7a891a8cdb1ab7207abf59/public/app/features/manage-dashboards/components/ImportDashboardOverview.tsx#L90

I can't remember why tbh, I believe @dprokop suggested this.

Reply:
OK ok, I would understand it if we want to reuse the content of the form but I guess it is only used in one place so that can't be it 🙈